src/value.h

```diff
@@ -137,9 +134,11 @@ typedef struct RSValue {
 } RSValue;
 #pragma pack()
 
-#define RS_DUOVAL_VAL(v) ((v).duoval.vals[0])
-#define RS_DUOVAL_OTHERVAL(v) ((v).duoval.vals[1])
-#define RS_DUOVAL_OTHER2VAL(v) ((v).duoval.vals[2])
+#define RS_JSONVAL_ITER(v) ((v).jsonval->iter)
+#define RS_JSONVAL_FIRST(v) ((v).jsonval->first)
+#define RS_JSONVAL_SERIALIZED(v) ((v).jsonval->serialized)
+#define RS_JSONVAL_EXPANDED(v) ((v).jsonval->expanded)
```

Comment:
nitpicking, but since we usually handle `RSValue`s with pointers, consider making these macros suitable for pointers